apps/web/src/pages/auth/InvitationPage.tsx

```diff
@@ -11,43 +11,48 @@ import { SignUpForm } from '../../components/auth/SignUpForm';
 import { colors, Text, Button } from '../../design-system';
 import { AuthContext } from '../../store/authContext';
 import { useAcceptInvite } from '../../components/auth/use-accept-invite.hook';
-import { When } from '../../components/utils/When';
+import { LoginForm } from '../../components/auth/LoginForm';
 
 export default function InvitationPage() {
+  const queryClient = useQueryClient();
   const navigate = useNavigate();
   const { token, logout, currentUser } = useContext(AuthContext);
   const location = useLocation();
   const isLoggedIn = !!token;
-  const { token: tokenParam } = useParams<{ token: string }>();
-  const { isLoading: loadingAcceptInvite, submitToken } = useAcceptInvite();
+  const { token: invitationToken } = useParams<{ token: string }>();
+  const tokensRef = useRef({ token, invitationToken });
+  tokensRef.current = { token, invitationToken };
+  const { isLoading: isAcceptingInvite, submitToken } = useAcceptInvite();
   const { data, isInitialLoading } = useQuery<IGetInviteResponseDto, IGetInviteResponseDto>(
     ['getInviteTokenData'],
-    () => getInviteTokenData(tokenParam || ''),
+    () => getInviteTokenData(invitationToken || ''),
     {
-      enabled: !!tokenParam,
+      enabled: !!invitationToken,
+      refetchOnWindowFocus: false,
     }
   );
   const inviterFirstName = data?.inviter?.firstName || '';
   const organizationName = data?.organization.name || '';
-
-  const existingUser = tokenParam && data?._userId;
-  const invalidCurrentUser = existingUser && currentUser && currentUser._id !== data?._userId;
-
-  const acceptToken = async () => {
-    if (existingUser && currentUser && currentUser._id === data?._userId && isLoggedIn) {
-      const result = await submitToken(tokenParam as string, true);
-      if (result) navigate('/templates');
-    }
-  };
+  const existingUser = !!(invitationToken && data?._userId);
+  const isLoggedInAsInvitedUser = !!(isLoggedIn && existingUser && currentUser && currentUser._id === data?._userId);
```

Comment:
```suggestion
  const isLoggedInAsInvitedUser = !!(isLoggedIn && existingUser && currentUser?._id === data?._userId);
```
I think checking `currentUser` on its own is superflous as part of this seeing the last branch of the condition. But this flag calculation is very brittle, so I might be wrong.

Reply:
it should stay there, otherwise, if remove it, the last branch might compare `undefined === data?._userId` which is not what we want